tests/cnf/core/network/metallb/tests/bgp-connect-time-tests.go

```diff
@@ -0,0 +1,224 @@
+		Expect(err).ToNot(HaveOccurred(), "Failed to validate metalLb exported ip address")
+
+		By("Listing master nodes")
+		masterNodeList, err = nodes.List(APIClient,
+			metav1.ListOptions{LabelSelector: labels.Set(NetConfig.ControlPlaneLabelMap).String()})
+		Expect(err).ToNot(HaveOccurred(), "Fail to list master nodes")
+		Expect(len(masterNodeList)).To(BeNumerically(">", 0),
+			"Failed to detect master nodes")
+	})
+
+	BeforeEach(func() {
+		By("Creating a new instance of MetalLB Speakers on workers")
+		err := metallbenv.CreateNewMetalLbDaemonSetAndWaitUntilItsRunning(tsparams.DefaultTimeout, workerLabelMap)
+		Expect(err).ToNot(HaveOccurred(), "Failed to recreate metalLb daemonset")
+
+		_, err = pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+			LabelSelector: tsparams.FRRK8sDefaultLabel,
+		})
+		Expect(err).ToNot(HaveOccurred(), "Fail to list speaker pods")
+
+		By("Collecting information before test")
+		frrk8sPods, err = pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+			LabelSelector: tsparams.FRRK8sDefaultLabel,
+		})
+		Expect(err).ToNot(HaveOccurred(), "Failed to list speaker pods")
+
+	})
+
+	AfterEach(func() {
+		By("Cleaning MetalLb operator namespace")
+		metalLbNs, err := namespace.Pull(APIClient, NetConfig.MlbOperatorNamespace)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull metalLb operator namespace")
+		err = metalLbNs.CleanObjects(
+			tsparams.DefaultTimeout,
+			metallb.GetBGPPeerGVR(),
+			metallb.GetBFDProfileGVR(),
+			metallb.GetBGPPeerGVR(),
+			metallb.GetBGPAdvertisementGVR(),
+			metallb.GetIPAddressPoolGVR(),
+			metallb.GetMetalLbIoGVR())
+		Expect(err).ToNot(HaveOccurred(), "Failed to remove object's from operator namespace")
+
+		By("Cleaning test namespace")
+		err = namespace.NewBuilder(APIClient, tsparams.TestNamespaceName).CleanObjects(
+			tsparams.DefaultTimeout,
+			pod.GetGVR(),
+			service.GetServiceGVR(),
+			configmap.GetGVR(),
+			nad.GetGVR())
+		Expect(err).ToNot(HaveOccurred(), "Failed to clean test namespace")
+	})
+
+	It("Verify configuration of a FRR node router peer with the connectTime less than the default of 120 seconds",
+		reportxml.ID("74414"), func() {
+
+			By("Creating BGP Peers with 10 second retry connect timer")
+			createBGPPeerWithConnectTimeAndVerifyIfItsReady(metav1.Duration{Duration: 10 * time.Second},
+				ipv4metalLbIPList[0], "", 64500, false, frrk8sPods)
+
+			By("Validate BGP Peers with 10 second retry connect timer")
+			Eventually(func() string {
+				// Get the routes
+				connectTime, err := frr.VerifyBGPConnectTime(frrk8sPods, ipv4metalLbIPList[0], 10)
+				Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP connect time")
+
+				// Return the routes to be checked
+				return connectTime
+			}, 60*time.Second, 5*time.Second).Should(ContainSubstring("ConnectRetryTimer: 10"),
+				"Fail to find received BGP connect time")
+		})
```

Comment:
I think it would be cleaner if the function returns the connect time and the eventually verifies it is the intended one (here it seems there is some sort of duplication)

Reply:
The func VerifyBGPConnectTime runs the command "show bgp neighbor %s json", bgpPeerIP and returns the connectTimer from the json ConnectRetryTimer: <value>. The Eventual waits for bgp to be established then checks the substring.  So not sure of the duplication? Thanks

Reply:
Ok I see what you are talking about. I changed the name of VerifyBGPConnectTime to FetchBGPConnectTimeValue it only returns the int value. Then the eventual checks that it is the correct value